fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -305,6 +313,12 @@ def __init__(
         init_start = time.time()
         super().__init__()
         self.process_group = process_group or get_process_group_cached()
+        if process_group_reduce_scatter == ProcessGroupName.all_gather:
```

Comment:
lets add a comment about what this is doing.

Reply:
done